deeplearning4j-keras/src/main/java/org/deeplearning4j/keras/DeepLearning4jEntryPoint.java

```diff
@@ -1,42 +1,386 @@
+                fitParams.getTrainXPath(),
+                fitParams.getTrainYPath()
+            );
+
+            if(fitParams.isDoValidation()) {
+                testIterator = new HDF5MiniBatchDataSetIterator(
+                    fitParams.getValidationXPath(),
+                    fitParams.getValidationYPath()
+                );
+            } else {
+                testIterator = null;
+            }
+
+            // loop through specified epochs
+            for (int i = 0; i < fitParams.getNbEpoch(); i++) {
+                log.info("Fitting epoch: " + i);
+
+                model.fit(trainIterator);
+                double score = model.score();
+                log.info("Score is "+score);
+
+                if(fitParams.isDoValidation()) {
+                    log.info("Evaluating epoch: " + i);
+                    model.evaluate(testIterator);
+                }
+            }
+
+            log.info("model.fit() operation complete.");
 
+        } catch (Throwable e) {
+            log.error("Error while performing model.fit()", e);
+            throw e;
+        }
+    }
+
+    /**
+     * Evaluates a model using a given dataset.
+     *
+     * @param evaluateParams Parameters for a Keras evaluate operation
+     */
+    public void sequentialEvaluate(EvaluateParams evaluateParams) throws Exception {
         try {
-            MultiLayerNetwork multiLayerNetwork = neuralNetworkReader.readNeuralNetwork(entryPointFitParameters);
+            MultiLayerNetwork model = evaluateParams.getSequentialModel();
 
             DataSetIterator dataSetIterator = new HDF5MiniBatchDataSetIterator(
-                    entryPointFitParameters.getTrainFeaturesDirectory(),
-                    entryPointFitParameters.getTrainLabelsDirectory()
+                evaluateParams.getFeaturesDirectory(),
+                evaluateParams.getLabelsDirectory()
             );
 
-            for (int i = 0; i < entryPointFitParameters.getNbEpoch(); i++) {
-                log.info("Fitting: " + i);
+            model.evaluate(dataSetIterator);
+
+            log.info("model.evaluate() operation complete.");
 
-                multiLayerNetwork.fit(dataSetIterator);
+        } catch (Throwable e) {
+            log.error("Error while performing model.evaluate()", e);
+            throw e;
+        }
+    }
+
+    /**
+     * Predict the label of a single feature.
+     *
+     * @param predictParams A single feature and associated parameters
+     */
+    public void sequentialPredict(PredictParams predictParams) throws Exception {
+        try {
+            MultiLayerNetwork model = predictParams.getSequentialModel();
+
+            DataSetIterator dataSetIterator = new HDF5MiniBatchDataSetIterator(
+                predictParams.getFeaturesDirectory()
+            );
+
+            while(dataSetIterator.hasNext()) {
+                DataSet data = dataSetIterator.next();
+                model.output(data.getFeatures());
```

Comment:
Currently results are output to the console. I'm unsure of how what we should be doing with them. I can return an object, but it will be an INDArray and not python-friendly. Ideas?